libitm/libitm.h

```diff
@@ -45,6 +45,11 @@ extern "C" {
 
 #define ITM_NORETURN	__attribute__((noreturn))
 #define ITM_PURE __attribute__((transaction_pure))
+#if defined(__i386__) || defined(__x86_64__)
```

Comment:
Why is this not needed for AArch64? Have you find any reference that this ABI should not be supported on AArch64? Furthermore this change won't be accepted upstream as it affects non-Cygwin targets.

Reply:
This is a fix for the patch
https://github.com/Windows-on-ARM-Experiments/gcc-woarm64/pull/38/commits/cadd6550817f676e2c814ee51aacda225098d058#diff-159e3a4bab15c66a63dda9e0959abbef336051d38fa391c3c295d73af4ae3ce8R48
It looks like this abi is supported only by ix86

Reply:
I can guess but do you have some evidence for 

> It looks like this abi is supported only by ix86
?

I can for example find https://github.com/ARM-software/abi-aa/blob/main/sysvabi64/sysvabi64.rst

Reply:
It is not implemented yet for aarch64 on any target.

Reply:
I asked Cygwin developers on IRC, and from their answer I'd assume that this is either deprecated or nice-to-have feature. I'd exclude the entire https://github.com/Windows-on-ARM-Experiments/gcc-woarm64/pull/38/commits/cadd6550817f676e2c814ee51aacda225098d058 patch  then.

Reply:
It will be merged in this PR as it was discussed offline and refactored later.